### PR TITLE
Added payment view and payment method

### PR DESCRIPTION
**Pull request by KAVAN-DESAI**

### CODE REVIEW

File: lib/api/cln/cln_client.dart [hunk 1]
```diff
@@ -109,4 +112,29 @@ class CLNApi extends AppApi {
...
   }
+
+  @override
+  Future<AppPayInvoice> payInvoice({required String invoice, int? msat}) async {
+    dynamic params;
+    switch (mode) {
+      case ClientMode.grpc:
+        if (msat == -1) {
+          params = CLNPayRequest(grpcRequest: PayRequest(bolt11: invoice));
+        } else {
```
Comment (vincenzopalazzo) [inline lib/api/cln/cln_client.dart:123]:
```suggestion
        if (msat == -1) {
          params = CLNPayRequest(grpcRequest: PayRequest(bolt11: invoice));
        } else {
```

Why this -1 here? why not using the null value directly

File: lib/views/pay/pay_view.dart [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      return error;
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    amountMsat = -1;
+    boltString = "";
+    paymentResponse = null;
```
Comment (vincenzopalazzo) [inline lib/views/pay/pay_view.dart:39]: I like the idea to init the state inside `initState` and not inside the constructor, but I was wondering if the init state trigger something that required that `amountMsat` will be initialized?

File: pubspec.yaml [hunk 3]
```diff
@@ -11,35 +11,25 @@ dependencies:
...
-    git:
-      url: https://github.com/dart-lightning/lndart.clightning.git
-      path: packages/rpc
+  cln_grpc: ^0.0.1-beta.2
+  cln_common: ^0.0.1-beta.1
+  clightning_rpc: 0.0.2-beta.1
   trash_component:
     git:
       url: https://github.com/vincenzopalazzo/trash
       path: packages/components
```
Comment (vincenzopalazzo) [inline pubspec.yaml:20]: we can move also this in the pubdev version? we release the version with your change, right?

File: pubspec.yaml [hunk 4]
```diff
@@ -11,35 +11,25 @@ dependencies:
...
   flutter_test:
     sdk: flutter
   flutter_lints: ^2.0.0
 
 dependency_overrides:
-  cln_common:
-    git:
-      url: https://github.com/dart-lightning/lndart.clightning.git
-      path: packages/cln_common
+  cln_common: ^0.0.1-beta.1
```
Comment (vincenzopalazzo) [inline pubspec.yaml:32]: this can be just commented out, is used to solve the git recursion problem

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: lib/views/pay/pay_view.dart (hunk 2 above)
Comment (KAVAN-DESAI) [reply lib/views/pay/pay_view.dart:39]: Yeah will make `amountMsat` nullable

File: pubspec.yaml (hunk 3 above)
Comment (KAVAN-DESAI) [reply pubspec.yaml:20]: ooh super cool

**Pull request merged**